wormchain/interchaintest/cw_wormhole_test.go

```diff
@@ -0,0 +1,482 @@
+package ictest
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"testing"
+
+	"github.com/docker/docker/client"
+	"github.com/strangelove-ventures/interchaintest/v4"
+	"github.com/strangelove-ventures/interchaintest/v4/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v4/ibc"
+	"github.com/strangelove-ventures/interchaintest/v4/testutil"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/wormhole-foundation/wormchain/interchaintest/guardians"
+	"github.com/wormhole-foundation/wormchain/interchaintest/helpers"
+	"github.com/wormhole-foundation/wormchain/interchaintest/helpers/cw_wormhole"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+)
+
+func createSingleNodeCluster(t *testing.T, wormchainVersion string, guardians guardians.ValSet) []ibc.Chain {
+	numWormchainVals := len(guardians.Vals)
+	numFullNodes := 0
+
+	wormchainConfig.Images[0].Version = wormchainVersion
+	wormchainConfig.ModifyGenesis = ModifyGenesis(votingPeriod, maxDepositPeriod, guardians, true)
+
+	cf := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		{
+			ChainName:     "wormchain",
+			ChainConfig:   wormchainConfig,
+			NumValidators: &numWormchainVals,
+			NumFullNodes:  &numFullNodes,
+		},
+	})
+
+	// Get chains from the chain factory
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+
+	return chains
```

Comment:
i think it makes sense to return a single chain here instead of []ibc.Chain, to make it easier to consume in the tests